user-guide/modules/ROOT/pages/task-metaprogramming.adoc

```diff
@@ -1,4 +1,40 @@
 = Metaprogramming
 :navtitle: Metaprogramming
 
-Content under construction
\ No newline at end of file
+Boost provides several libraries that are highly useful in metaprogramming, and can greatly ease the process of building your own metaprogramming-based pass:[C++] library.
+
+Metaprogramming in pass:[C++] is a complex topic that can lead to complex code. It should be used judiciously, as it can make code more difficult to understand and maintain. Also, modern pass:[C++] (pass:[C++]11 and later) provides many features, such as `constexpr` and https://en.wikipedia.org/wiki/Variadic_template[variadic templates], that can achieve at runtime what was previously only possible with metaprogramming, so make sure to familiarize yourself with these features before delving too deeply into metaprogramming.
+
+== Libraries
+
+[circle]
+* boost:mpl[]:  A Metaprogramming library that provides a framework of compile-time algorithms, sequences and metafunctions, that can greatly facilitate metaprogramming tasks.
+
+* boost:fusion[]:  Provides a set of tools for working with tuples, which can be used to work with heterogeneous collections at compile time. It helps to bridge the gap between the compile-time world and the runtime world.
+
+* boost:hana[]:  This is a modern, high-level library for metaprogramming. Like boost:fusion[], it is used for working with heterogeneous collections, but it uses modern pass:[C++] features and can be easier to use than (and can be considered the successor to) both boost:mpl[] and boost:fusion[].
+
+* boost:type-traits[]:  Provides a collection of templates for information about types. This can be useful in many metaprogramming tasks, as it allows for compile-time introspection of types.
+
+* boost:static-assert[]:  Provides a macro for compile-time assertions. This can be useful in metaprogramming to enforce constraints at compile time.
+
+* boost:preprocessor[]:  Provides tools for manipulating preprocessor directives, which can be useful in certain metaprogramming tasks.
+
+== Metaprogramming Applications
+
+Metaprogramming, the practice of writing code that generates or manipulates other code, has several powerful applications, particularly in languages like pass:[C++] that support compile-time metaprogramming. Be careful though, despite its power, metaprogramming can also lead to complex, hard-to-understand code and should be used judiciously. Here are some of the primary applications of metaprogramming:
+
+[circle]
```

Comment:
We should probably talk to Spencer about this. Bullet lists should be rendered without [circle]. [circle] should only be needed to customize the behavior.